beacon_node/rest_api/src/beacon.rs

```diff
@@ -123,6 +128,50 @@ pub fn get_block_root<T: BeaconChainTypes>(
     ResponseBuilder::new(&req)?.body(&root)
 }
 
+fn make_sse_response_chunk(new_head_hash: SignedBeaconBlockHash) -> std::io::Result<Chunk> {
+    let mut buffer = Vec::new();
+    {
+        let mut sse_message = uhttp_sse::SseMessage::new(&mut buffer);
+        let untyped_hash: Hash256 = new_head_hash.into();
+        write!(sse_message.data()?, "{:?}", untyped_hash)?;
+    }
+    let chunk: Chunk = buffer.into();
+    Ok(chunk)
+}
+
+pub fn stream_forks<T: BeaconChainTypes>(
+    log: Logger,
+    mut events: BusReader<SignedBeaconBlockHash>,
+) -> ApiResult {
+    let (mut sender, body) = Body::channel();
+    std::thread::spawn(move || {
```

Comment:
Seems a bit of a shame that each SSE listener requires a new OS thread, I thought perhaps we might be able to utilize the tokio threadpool for this? I haven't looked into it though, perhaps that's infeasible?

If I had to guess, I'd say we could do this if we used a futures-aware [tokio channel](https://docs.rs/tokio/0.1.22/tokio/sync/mpsc/fn.channel.html) instead of the `bus` and we had access to a tokio executor then we could spawn a task instead of a thread.

Reply:
As stated in the PR description, this is just a first version.  If we need to scale to larger number of connections, we can multiplex all events onto a single background thread pretty easily.  Plus `Bus` doesn't block and supports multiple receivers whereas tokio channel supports only one receiver and (I surmise) blocks.